From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: ByThePowerOfScience
 <16433721+ByThePowerOfScience@users.noreply.github.com>
Date: Fri, 23 Feb 2024 15:45:33 -0500
Subject: [PATCH] 6249: Fixed false error in GhidraJarBuilder.java

Changed from "only giving an error if we WERE able to write to the sources jar" to "only giving an error if we were NOT able to write to the sources jar".

Moved zip.close out of if-else tree.

Wrapped zip.close in try-catch block
---
 .../src/main/java/ghidra/util/GhidraJarBuilder.java  | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java b/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java
index ef2ade34fe..db4f94021e 100644
--- a/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java
+++ b/Ghidra/Features/Base/src/main/java/ghidra/util/GhidraJarBuilder.java
@@ -338,12 +338,16 @@ public class GhidraJarBuilder implements GhidraLaunchable {
 				wroteToZip |= writeZipRecursively(zip, srcDir.getAbsolutePath(), srcDir);
 			}
 		}
-		if (wroteToZip) {
-			System.out
-					.println("Can't create source zip!  Has source been downloaded and installed?");
-			// zip.close reports error if nothing has been written to it
+		if (!wroteToZip) {
+			System.out.println("Can't create source zip!  Has source been downloaded and installed?");
+		}
+		// zip.close reports error if nothing has been written to it
+		try {
 			zip.close();
+		} catch (IOException e) {
+			e.printStackTrace();
 		}
+
 	}
 
 	private void writeModuleSrcZipToOverallSrcZip(Zip zip, File srcZipFileForModule)
-- 
2.44.0

